### PR TITLE
Refactor(eos_designs): Wildcard dict to list l3_edge

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`l3_edge`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: (Claus)
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: (Tony)
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2 [hunk 1]
```diff
@@ -5,7 +5,10 @@
 {% set l3_edge_data.isis_interfaces = [] %}
 {% for p2p_link in l3_edge.p2p_links | arista.avd.natural_sort %}
 {%     if p2p_link.profile is arista.avd.defined and l3_edge.p2p_links_profiles is arista.avd.defined %}
-{%         set p2p_profile = l3_edge.p2p_links_profiles[p2p_link.profile] | arista.avd.default({}) %}
+{%         set p2p_profile = l3_edge.p2p_links_profiles | arista.avd.default() |
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2]: When the variable is not defined, the default value is set to **None/null**. So performing the next sequence of operations over the null value is throwing the error.

Same for below.

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md [hunk 2]
```diff
@@ -10,10 +10,11 @@ Make sure to configure the variables in a group_vars file covering all devices m
 ```yaml
 l3_edge:
   p2p_links_ip_pools:
-    < p2p_pool_name_1 >: < IPv4_address/Mask >
+    - pool_name: < p2p_pool_name_1 >
+      ip_address: < IPv4_address/Mask >
   p2p_links_profiles:
-    < p2p_profile_name >:
-      < any variable supported under p2p_links can be inherited from a profile >
+    - profile_name: < p2p_profile_name >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md]:
```suggestion
    - name: < p2p_profile_name >
```

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md [hunk 3]
```diff
@@ -10,10 +10,11 @@ Make sure to configure the variables in a group_vars file covering all devices m
 ```yaml
 l3_edge:
   p2p_links_ip_pools:
-    < p2p_pool_name_1 >: < IPv4_address/Mask >
+    - pool_name: < p2p_pool_name_1 >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md]:
```suggestion
    - name: < p2p_pool_name_1 >
```

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md [hunk 4]
```diff
@@ -10,10 +10,11 @@ Make sure to configure the variables in a group_vars file covering all devices m
...
+      ip_address: < IPv4_address/Mask >
   p2p_links_profiles:
-    < p2p_profile_name >:
-      < any variable supported under p2p_links can be inherited from a profile >
+    - profile_name: < p2p_profile_name >
+      variable: < any variable supported under p2p_links can be inherited from a profile >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md]:
```suggestion
      <key>: < any setting supported under p2p_links can be inherited from a profile >
```

File: ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2 (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2]: corrected, facing an error in CI

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md (hunk 2 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/doc/l3-edge-v4.0.md]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2 [hunk 5]
```diff
@@ -5,7 +5,10 @@
 {% set l3_edge_data.isis_interfaces = [] %}
 {% for p2p_link in l3_edge.p2p_links | arista.avd.natural_sort %}
 {%     if p2p_link.profile is arista.avd.defined and l3_edge.p2p_links_profiles is arista.avd.defined %}
-{%         set p2p_profile = l3_edge.p2p_links_profiles[p2p_link.profile] | arista.avd.default({}) %}
+{%         set p2p_profile = l3_edge.p2p_links_profiles | arista.avd.default({}) |
+                                                                      arista.avd.convert_dicts('name','<key>') |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2]: you should not have a secondary key on this. The <key> was just a description to tell the user what to do.

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2]: okay, thanks

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Comment by tgodaA** [comment]
We need to add molecule scenario for this as well.

**Comment by ClausHolbechArista** [comment]
We will add CI coverage in a separate PR. Issue #1792.

**Pull request merged**